pkg/networking/trafficplugin/registry/registry.go

```diff
@@ -105,8 +113,25 @@ func (esd *EdgeServiceDiscovery) FindMicroServiceInstances(consumerID, microServ
 			}
 		}
 	}
+	// set targetPort from endpoints if hostPort == 0 still
+	if hostPort == 0 {
+		for _, a := range eps.Subsets {
+			for _, port := range a.Ports {
+				if port.Port != 0 {
+					microServiceInstances = append(microServiceInstances, &registry.MicroServiceInstance{
+						InstanceID:   fmt.Sprintf("%s.%s|%s.%d", namespace, name, a.Addresses[0].IP, port.Port),
+						ServiceID:    fmt.Sprintf("%s#%s#%s", namespace, name, a.Addresses[0].IP),
+						HostName:     "",
+						EndpointsMap: map[string]string{proto: fmt.Sprintf("%s:%d", a.Addresses[0].IP, port.Port)},
+					})
+				}
+			}
+		}
+	}
+
 	for _, p := range pods {
-		if p.Status.Phase == v1.PodRunning {
+		// set Pod ip if hostPort != 0
+		if p.Status.Phase == v1.PodRunning && hostPort != 0 {
```

Comment:
why not just use 
```
if hostPort == 0 {
} else {
   for _, p := range pods {
        if p.Status.Phase == v1.PodRunning {
  }
}
```

Reply:
Yes, you are right, I will modify it immediately.